include/orc/dwarf_structs.hpp

```diff
@@ -184,37 +184,45 @@ const char* to_string(arch arch);
 // use the same abbreviation, but because the die is listed in a different place in the debug_info
 // data block, it's values will be different than previous "stampings" of the abbreviation.
 struct die {
-    std::uint32_t _debug_info_offset{0}; // relative from top of __debug_info
-    arch _arch{arch::unknown};
-    std::uint32_t _abbrev_code{0};
-    dw::tag _tag{dw::tag::none};
-    dw::has_children _has_children{dw::has_children::no};
+    // Because the quantity of these created at runtime can beon the order of millions of instances,
+    // these are ordered for optimal alignment. If you change the ordering, or add/remove items
+    // here, please consider alignment issues.
     pool_string _object_file;
     std::string _path;
-    std::vector<attribute> _attributes;
-    bool _type_resolved{false};
+    attribute* _attributes;
+    std::size_t _attributes_size{0};
     std::size_t _hash{0};
+    std::uint32_t _debug_info_offset{0}; // relative from top of __debug_info
+    dw::tag _tag{dw::tag::none};
+    arch _arch{arch::unknown};
+    bool _has_children{false};
+    bool _type_resolved{false};
+
+    auto begin() { return _attributes; }
```

Comment:
This foursome of routines let us use the `die` in `for` loops, and it'll go over the die's attributes:

```c++
for (const auto& attribute : my_die) {
    //...
}
```

I'm not super-sold on this yet, and may add an accessor struct to make things a little more clear. The end result would look like:

```c++
for (const auto& attribute : my_die.attributes()) {
    //...
}
```